### PR TITLE
Webcam widget

### PR DESCRIPTION
**Pull request by nontas**
This PR adds a webcam widget function (thanks to @patricksnape major contribution!!)

The widget is able to take multiple snapshots and return the captured images as a `list` of `Image` objects. It consists of the following buttons: "Take screenshot", "Close" and a resolution slider. The widget also has a panel at the bottom that visualizes the captured snapshots as thumbnails. This preview panel works as a FIFO stack and the user can determine the number of thumbnails to use.

The widget can be called as:

``` python
from menpowidgets import webcam_widget

images = webcam_widget()
```

**TODO**
- [x] Make sure that the docs build correctly. There was something going wrong with `sphinx` on my laptop.

### CODE REVIEW

File: menpowidgets/js/webcam.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+require(["nbextensions/widgets/widgets/js/widget", "nbextensions/widgets/widgets/js/manager"],
```
Comment (patricksnape) [inline menpowidgets/js/webcam.js]: @jabooth whatever you do don't look at this JS code :scream:

**Comment by patricksnape** [comment]
Docs build for me. Also everything now seems to work for me. :+1: on my end.

**Comment by jabooth** [comment]
@patricksnape @nontas great!

Only comment is that it doesn't work in safari (which I tried first).

http://caniuse.com/#feat=stream

can we:
1. Add a note about this to the docstring
2. Check `if (getUserMedia === undefined)` in the JS and then print a message saying:

> Unfortunately, this browser is not supported by the webcam widget. Try Chrome or Firefox.

**Comment by patricksnape** [comment]
jenkins retest this please

**Comment by patricksnape** [comment]
jenkins retest this please

**Comment by patricksnape** [comment]
OK going to merge this once travis passes.

**Pull request merged**